plugins/leemons-plugin-users/frontend/src/pages/protected/SelectProfile/index.js

```diff
@@ -105,28 +105,55 @@ export default function SelectProfile({ session }) {
     }
   }
 
+  /**
+   * This function initializes the profile selection process
+   */
   async function init() {
+    // Sets the layout state to not private and profile not checked
     setLayoutState({ ...layoutState, private: false, profileChecked: false });
+    // Fetches centers, remembered login, and user token concurrently
     const [{ centers }, { profile, center }, userToken] = await Promise.all([
       getUserCentersRequest(),
       getRememberLoginRequest(getCookieToken()),
       leemons.api(`v1/users/users`),
     ]);
 
-    store.centers = centers;
+    // Filters out denied profiles from each center and removes centers with no allowed profiles
+    const deniedProfiles = deploymentConfig?.deny?.profiles || [];
+    const processedCenters = centers.map((centre) => ({
+      ...centre,
+      profiles: centre.profiles.filter(
+        (userProfile) => !deniedProfiles.includes(userProfile.sysName)
```

Comment:
n^m complexity, much better if deniedProfiles converted to object and then check for key existance in the object. New complexity of n+m :P